File: norden/norden/report/report_purchase_order/report_purchase_order.py
# Copyright (c) 2023, Teampro and contributors
# For license information, please see license.txt


import frappe
from frappe import _


def execute(filters=None):
	columns=get_columns(filters)
	data = get_data(filters)
	return columns, data

def get_columns(filters):
	columns = []
	columns += [
		_("PO NO") + ":Link/Purchase Order:110",
		_("File NO") + ":Data:100",
		_("Title") + ":Data:100",
		_("Vendor") + ":Data:100",
		_("Order Date") + ":Data:100",
		_("Status") + ":Data:100",
		_("Grand Total") + ":Data:100",
		_("Grand Total (BC)") + ":Data:110"
	]
	return columns

def get_conditions(filters):
	conditions = ""
	if filters.get("from_date") and filters.get("to_date"):
		conditions += " transaction_date between %(from_date)s and %(to_date)s"
	if filters.get("supplier"):
		conditions += " and supplier= %(supplier)s"
	if filters.get("status"):
		conditions += " and status = %(status)s"
	
	return conditions, filters

def get_data(filters):
	data = []
	conditions, filters = get_conditions(filters)
	sa = []
	sa = frappe.db.sql("""select * from `tabPurchase Order` where %s """%conditions, filters,as_dict=True)
	if sa:
		for i in sa:
			row=[i.name,i.file_number,i.title,i.supplier,i.transaction_date,i.status,i.grand_total,i.base_grand_total]
			data.append(row)
			frappe.errprint(row)
	return data

File: norden/norden/report/report_purchase_order/report_purchase_order.js
// Copyright (c) 2023, Teampro and contributors
// For license information, please see license.txt


frappe.query_reports["Report Purchase Order"] = {
	"filters": [
	
		{
			"fieldname": "from_date",
            "label": __(" Order From Date"),
            "fieldtype": "Date",
			"reqd": 1
		},
		{
	
			"fieldname": "to_date",
			"label": __("Order To Date"),
			"fieldtype": "Date",
			"reqd": 1
		},
		{
			"fieldname": "supplier",
			"fieldtype": "Link",
			"options": "Supplier",
			"label": __("Vendor"),
			"width": "50px"
        },
		{
			"fieldname": "project",
			"fieldtype": "Link",
			"options": "Project",
			"label": __("Project"),
			"width": "50px"
        },
		{
			"fieldname":"status",
			"label":__("Status"),
			"fieldtype":"Select",
			"options":["","Draft","On Hold","To Receive and Bill","To Bill",
			"To Receive",
			"Completed",
			"Cancelled",
			"Closed",
			"Delivered"],
			"default": "",
			"width": "100px"
		}
	]
};
